xapian-core/tests/api_termgen.cc

```diff
@@ -140,6 +140,11 @@ static const test test_simple[] = {
     { "", "극지라", "극[1] 극지:1 라[3] 지[2] 지라:1" },
     { "", "ウルス アップ", "ア[4] アッ:1 ウ[1] ウル:1 ス[3] ッ[5] ップ:1 プ[6] ル[2] ルス:1" },
 
+    // Test parsing mixed CJK numbers (only with Chinese now)
+    { "", "2千3百", "2千3百[1]"},
+    { "", "有2千三百", "2千三百[2] 有[1]"},
+    { "", "我有2千3百零一块钱", "2千3百零一[3] 块[4] 块钱:1 我[1] 我有:1 有[2] 钱[5]"},
+
```

Comment:
These are good tests for the cases you want the patch to handle.

But we should also ensure that it doesn't handle cases which it shouldn't.  For example, if you intend this patch to only affect the ASCII digits, then there should be tests that to check what happens to cases like the above but with `'2'` and/or `'3'` replaced by Chinese digits, Arabic-Indic digits, etc.